probe/probe.go

```diff
@@ -302,3 +329,347 @@ func probeNode(c *k8s.Client) {
+		Get().
+		Namespace("kube-system").
+		Resource("pods").
+		Name(pods.Items[0].Name).
+		SubResource("exec").
+		VersionedParams(&corev1.PodExecOptions{
+			Container: pods.Items[0].Spec.Containers[0].Name,
+			Command:   cmdArr,
+			Stdin:     true,
+			Stdout:    true,
+			Stderr:    true,
+			TTY:       false,
+		}, scheme.ParameterCodec)
+	exec, err := remotecommand.NewSPDYExecutor(c.Config, "POST", req.URL())
+
+	if err != nil {
+		return err
+	}
+	go func() {
+		defer outStream.Close()
+		err = exec.Stream(remotecommand.StreamOptions{
+			Stdin:  os.Stdin,
+			Stdout: outStream,
+			Stderr: os.Stderr,
+			Tty:    false,
+		})
+	}()
+	buf, err := io.ReadAll(reader)
+	if err != nil {
+		log.Println("an error occured when reading file", err)
+		return err
+	}
+	var json = jsoniter.ConfigCompatibleWithStandardLibrary
+	err = json.Unmarshal(buf, &kd)
+	if err != nil {
+		log.Println("an error occured when parsing file", err)
+		return err
+	}
+	printKubeArmorProbeOutput(kd)
+	return nil
+}
+
+func printKubeArmorProbeOutput(kd *KubeArmorProbeData) {
+
+	fmt.Printf("\t OS Image: ")
+	color.Green(kd.OSImage)
+	fmt.Printf("\t Kernel Version: ")
+	color.Green(kd.KernelVersion)
+	fmt.Printf("\t Kubelet Version: ")
+	color.Green(kd.KubeletVersion)
+	fmt.Printf("\t Container Runtime: ")
+	color.Green(kd.ContainerRuntime)
+	fmt.Printf("\t Active LSM: ")
+	color.Green(kd.ActiveLSM)
+	fmt.Printf("\t Host Security: ")
+	color.Green(strconv.FormatBool(kd.HostSecurity))
+	fmt.Printf("\t Container Security: ")
+	color.Green(strconv.FormatBool(kd.ContainerSecurity))
+	fmt.Printf("\t Container Default Posture: \n")
+	fmt.Printf("\t\t File Action: ")
+	color.Green(kd.ContainerDefaultPosture.FileAction)
+	fmt.Printf("\t\t Capabilities Action: ")
+	color.Green(kd.ContainerDefaultPosture.CapabilitiesAction)
+	fmt.Printf("\t\t Network Action: ")
+	color.Green(kd.ContainerDefaultPosture.NetworkAction)
+	fmt.Printf("\t Host Default Posture: \n")
+	fmt.Printf("\t\t File Action: ")
+	color.Green(kd.HostDefaultPosture.FileAction)
+	fmt.Printf("\t\t Capabilities Action: ")
+	color.Green(kd.HostDefaultPosture.CapabilitiesAction)
+	fmt.Printf("\t\t Network Action: ")
+	color.Green(kd.HostDefaultPosture.NetworkAction)
+}
+
+// sudo systemctl status kubearmor
+func isSystemdMode() bool {
+	cmd := exec.Command("systemctl", "status", "kubearmor")
+	_, err := cmd.CombinedOutput()
+	if err != nil {
+		if _, ok := err.(*exec.ExitError); ok {
+			return false
+		} else {
+			return false
+		}
+	}
+	color.Green("\nFound KubeArmor running in Systemd mode \n\n")
+	return true
+}
+
+func probeSystemdMode() error {
+	var kd *KubeArmorProbeData
+	jsonFile, err := os.Open("/tmp/karmorProbeData.cfg")
+	if err != nil {
+		log.Println(err)
+		return err
+	}
+
+	buf, err := io.ReadAll(jsonFile)
+	if err != nil {
+		log.Println("an error occured when reading file", err)
+		return err
+	}
+	var json = jsoniter.ConfigCompatibleWithStandardLibrary
+	err = json.Unmarshal(buf, &kd)
```

Comment:
because the json response from kubearmor is an iterated json and the normal json unmarshal wasnt working for the unmarshalling, so in found that library as a solution